src/callbacks_step/stepsize.jl

```diff
@@ -99,15 +143,30 @@ function (cb::DiscreteCallback{Condition, Affect!})(ode::ODEProblem) where {Cond
                                                                             StepsizeCallback
                                                                             }
     stepsize_callback = cb.affect!
-    @unpack cfl_number = stepsize_callback
+    @unpack cfl_convective, cfl_diffusive = stepsize_callback
     u_ode = ode.u0
     t = first(ode.tspan)
     semi = ode.p
     mesh, equations, solver, cache = mesh_equations_solver_cache(semi)
     u = wrap_array(u_ode, mesh, equations, solver, cache)
 
-    return cfl_number *
-           max_dt(u, t, mesh, have_constant_speed(equations), equations, solver, cache)
+    dt_convective = cfl_convective *
+                    max_dt(u, t, mesh, have_constant_speed(equations), equations,
+                           solver, cache)
+
+    # Check if diffusive CFL should be considered.
+    # NOTE: 
+    # For non-zero `cfl_diffusive`, `semi` is expected to be a `SemidiscretizationHyperbolicParabolic`.
+    if cfl_diffusive > 0.0
```

Comment:
This is a bit of a hack, as this would crash if someone supplies `cfl_diffusive` for a simulation without parabolic equations.

I first tried another if-clause `if hasfield(semi, :equations_parabolic)` which unfortunately gave errors in some particular cases.